src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -37,6 +40,19 @@ protected override bool ReleaseHandle()
 
         internal sealed class SafeCurlMultiHandle : SafeHandle
         {
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+            private readonly List<int> _fdList = new List<int>();
```

Comment:
How many of these descriptors do we expect to have?  If it's a lot, and if it's common to add/remove them, would it be better to make this a HashSet or something where lookups are faster?  Or do we expect a small number such that we'd prefer the lower memory overhead of a list instead of the faster lookup?

Also, will it be common to have 0 descriptors?  If so, we should lazily-allocate the list.